h11/_headers.py

```diff
@@ -74,6 +74,11 @@ def normalize_and_validate(headers):
         # downstream." -- https://tools.ietf.org/html/rfc7230#section-3.2.4
         if name.strip() != name:
             raise LocalProtocolError("Illegal header name {}".format(name))
+        # "A sender MUST NOT generate a message that includes line folding"
+        # -- https://tools.ietf.org/html/rfc7230#section-3.2.4
```

Comment:
There is no other section.

The reason we want to forbid newlines is not because they're not RFC-valid: as @njsmith points out, they are. The reason we want to forbid them is because they can be used to conduct request smuggling attacks if users are in any way capable of inserting data into the header fields. This should be forbidden. Given that line folding is forbidden *anyway*, and that h11 does not allow users to control the way HTTP is framed (that's what h11 does!), the whitespace characters amount to framing characters and as such should be forbidden.